### DISCUSSION TITLE
Why project.json file not getting generate if project created by nx-dotnet plugin

### DISCUSSION THREAD

**Discussion by sshelake25** [nx-dotnet/nx-dotnet#902] [Q&A] [answered]
I create dotnet project using nx GUI. I don't get project.json file with required task for generate libs/webapi etc.

How to generate required project.json file for existing dotnet projects? I did long back I don;'t find that documenetation now.

![image](https://github.com/user-attachments/assets/6ef07321-6077-46ac-ac18-413492a70505)

**Comment by sshelake25** [comment]
https://www.nx-dotnet.com/docs/core/generators/import-projects  - Don't find documenation for this?

How to generate project.json file for existing / new nx dotnet projects?

**Reply by AgentEnder** [reply]
`project.json` isn't required for nx. If you are not able to build the project, its possible that the name is incorrect. Check `nx show projects` to see the list of all of your projects. In the example screenshot, I'd guess that the project is named `hello`, not `Hello` but I could be wrong.

To set a name for the project, you could add a `project.json` file with `{ "name": "foo" }`, but like I was saying that should be optional.

**Reply by sshelake25** [reply]
I have lot of dotnet projects in my mono repo, so project.json file not required? 
I think in "@nx-dotnet/core": "1.22.0", its was getting generated using generators - "import-projects".

Should I delete my project.json files from dotnet projects? In which cases I should use?

When I run frontend project test/build I am getting warnings 
Project 'aa-bb-cc-hello-pushnotifications' contains extension with invalid name (type).

**Reply by AgentEnder** [reply, marked as answer]
Thr extension warning is coming from angular CLI. I thought it had been fixed in the latest version of the plugin tbh, but I can look again later.

Like I said, project.json is optional. We changed to not generate it by default to better match how Nx works with project crystal enabled. You can add one to customize project settings if needed, but there's not a hard rule as to when you should or shouldn't do so.

**Reply by sshelake25** [reply]
Thank you for the updates @AgentEnder, It really helps. I have few more issue to report. I will open seprate threads
